src/Raven.Client/Documents/Subscriptions/SubscriptionWorkerOptions.cs

```diff
@@ -96,6 +96,7 @@ private SubscriptionWorkerOptions()
             Strategy = SubscriptionOpeningStrategy.OpenIfFree;
             MaxDocsPerBatch = 4096;
             TimeToWaitBeforeConnectionRetry = TimeSpan.FromSeconds(5);
+            WorkerStreamTimeout = TimeSpan.FromSeconds(30);
```

Comment:
Do you think that 'Worker` is important for the user? Maybe we could have something more generic - `ConnectionStreamTimeout` maybe?

Reply:
Additional question - how often do we sent heartbeats? I want to make sure that we won't get timeouts because of this setting when we are waiting long for new items

Reply:
Yeah that sounds better. Changed

Reply:
IIRC every 3 seconds

Reply:
So maybe we should only let the timeout be set for a value higher than the heartbeats time?

Reply:
That would make sense. Do we have 5 seconds setting hard coded or is it configurable?

Reply:
its 3 sec, hardcoded, I would take max of 15sec and WorkerStreamTimeout (so the min accpetable value is 15sec) on the server side.

Reply:
Waiting for new docs in server - heartbeat every 3 seconds
Waiting for worker to finish processing - heartbeat every 5 seconds
I think it should alert the user somehow that their timeout will be ignored
Maybe check that the given timeout is bigger than 15 secs when we are setting `ConnectionStreamTimeout` and throw if not

Reply:
Yeah, let's configure it to accept timeouts >= 15 seconds and throw otherwise. Please include the info and 15 s is the min value in the property docs xml comment.

Reply:
done